components/p3a/p3a_service.cc

```diff
@@ -236,18 +229,29 @@ void P3AService::OnHistogramChanged(const char* histogram_name,
     return;
   }
 
-  // Note that we store only buckets, not actual values.
-  size_t bucket = 0u;
-  const bool ok = samples->Iterator()->GetBucketIndex(&bucket);
-  if (!ok) {
-    LOG(ERROR) << "Only linear histograms are supported at the moment!";
-    NOTREACHED();
-    return;
-  }
+  // Assume that the sample == bucket for P3A measurements.
+  size_t bucket = std::max(sample, 0);
```

Comment:
While this is logical, it still feels a bit scary :D Are you sure we wouldn't get off by one errors for different type of histograms (e.g enumerations)?

Reply:
i understand the concern, but i think it should be okay. i tested the output of both the "exact linear" and "enumeration" and the values appeared as expected. could you maybe elaborate on your concern? i can do some more checks